docs/events/before-file-read.md

```diff
@@ -0,0 +1,77 @@
+# BEFORE_FILE_READ {#before-file-read}
+
+**sync | master | worker**
+
+The `BEFORE_FILE_READ` event is triggered before the test file is read to parse it. The event handler is executed synchronously. The event is also available in Testplane workers.
+
+## Subscription {#subscription}
+
+```javascript
+testplane.on(testplane.events.BEFORE_FILE_READ, ({ file, testplane, testParser }) => {
+    testParser.setController('<some-command-name>', {
+        '<some-helper>': function(matcher) {
+        // ...
+        }
+    });
+});
+```
+
+### Handler parameters {#handler-parameters}
+
+An object of the following format is passed to the event handler:
+
+```javascript
+{
+    file, // String: path to the file with the test
+    testplane, // Object: same as global.testplane
+    testParser // Object: type TestParserAPI
+}
+```
+
+### testParser: TestParserAPI {#test_parser}
```

Comment:
I see we are writing API reference here, so maybe test parser deserves it's own file